### DISCUSSION TITLE
FirestoreAdapter with NextJS

### DISCUSSION THREAD

**Discussion by bcthakre** [vercel/community#2633] [Help]
### Page to Investigate

https://authjs.dev/reference/adapter/firebase

### Steps to Reproduce

`npx create-next-app@latest my-next-app`

`npm install next-auth`

```
mkdir pages/api/auth
touch pages/api/auth/[...nextauth].js
```

**[...nextauth].js**
```
import NextAuth from "next-auth";
import GoogleProvider from "next-auth/providers/google";
import { FirestoreAdapter } from "@next-auth/firebase-adapter";
import { cert } from "firebase-admin/app";
import { db } from "../../../firebase";

export default NextAuth({
  providers: [
    GoogleProvider({
      clientId: process.env.GOOGLE_CLIENT_ID,
      clientSecret: process.env.GOOGLE_CLIENT_SECRET,
      
    })
  ],
  secret: process.env.JWT_SECRET,
  debug: true,
  adapter: FirestoreAdapter(db)
  // The following is not working wihh Vercel Deployment
  // adapter: FirestoreAdapter({
  //   credential: cert({
  //     projectId: process.env.NEXT_PUBLIC_FIREBASE_PROJECT_ID,
  //     clientEmail: process.env.FIREBASE_CLIENT_EMAIL,
  //     privateKey: process.env.FIREBASE_PRIVATE_KEY.replace(/\\n/g, "\n")
  //   })
  // })
});

```
firebase.js

```
import * as admin from 'firebase-admin';

if (!admin.apps.length) {
  admin.initializeApp({
    credential: admin.credential.cert({
      projectId: process.env.NEXT_PUBLIC_FIREBASE_PROJECT_ID,
      clientEmail: process.env.FIREBASE_CLIENT_EMAIL,
      privateKey: process.env.FIREBASE_PRIVATE_KEY.replace(/\\n/g, '\n'),
    }),
    databaseURL: process.env.FIREBASE_DATABASE_URL,
  });
}

const db = admin.firestore();
export { db };
```

The FirestoreAdapter works will on localhost but gives the following error after deployment on Vercel

**Try signing in with a different account.**

**Comment by bcthakre** [comment]
Apologies but neither of these FirestoreAdapter working in the above code.

**Comment by amyegan** [comment]
Hi @bcthakre. Have you verified that the needed [environment variables](https://vercel.com/docs/concepts/projects/environment-variables) are set for the `production` environment in your project? Typos and expired tokens are a common cause of problems when going from local development to `production`.

**Comment by bcthakre** [comment]
Thanks @amyegan  for looking into it. Rechecked the env. variables. They are mentioned as following and in all 3 environments i.e. Production, Preview, and Development

`
NEXT_PUBLIC_FIREBASE_PROJECT_ID=xyz

FIREBASE_CLIENT_EMAI=xyz@xyz.iam.gserviceaccount.com

FIREBASE_PRIVATE_KEY="-----BEGIN PRIVATE KEY-----LOREMIPSUM==\n-----END PRIVATE KEY-----\n"
`

I am curious if the FIREBASE_PRIVATE_KEY while mentioned in env variable should have the \n and all other letters and special character. Thoughts ? What else I could be missing.

**Reply by bcthakre** [reply]
Team,

Any help on this regard. Thanks for the help.
